engine/flink/model-util/src/main/scala/pl/touk/nussknacker/engine/flink/util/transformer/outer/OuterJoinTransformer.scala

```diff
@@ -0,0 +1,119 @@
+package pl.touk.nussknacker.engine.flink.util.transformer.outer
+
+import java.time.Duration
+import java.util.concurrent.TimeUnit
+
+import com.typesafe.scalalogging.LazyLogging
+import org.apache.flink.streaming.api.functions.TimestampAssigner
+import org.apache.flink.streaming.api.functions.co.CoProcessFunction
+import org.apache.flink.streaming.api.scala._
+import org.apache.flink.streaming.runtime.operators.windowing.TimestampedValue
+import pl.touk.nussknacker.engine.api._
+import pl.touk.nussknacker.engine.api.context.ProcessCompilationError.{CustomNodeError, NodeId}
+import pl.touk.nussknacker.engine.api.context.ValidationContext
+import pl.touk.nussknacker.engine.api.context.transformation._
+import pl.touk.nussknacker.engine.api.definition.{NodeDependency, OutputVariableNameDependency, Parameter, ParameterWithExtractor}
+import pl.touk.nussknacker.engine.flink.api.compat.ExplicitUidInOperatorsSupport
+import pl.touk.nussknacker.engine.flink.api.process.{FlinkCustomJoinTransformation, FlinkCustomNodeContext}
+import pl.touk.nussknacker.engine.flink.util.keyed.{StringKeyOnlyMapper, StringKeyedValue, StringKeyedValueMapper}
+import pl.touk.nussknacker.engine.flink.util.timestamp.TimestampAssignmentHelper
+import pl.touk.nussknacker.engine.flink.util.transformer.aggregate.{AggregateHelper, Aggregator}
+
+import scala.concurrent.duration.FiniteDuration
+
+class OuterJoinTransformer(timestampAssigner: Option[TimestampAssigner[TimestampedValue[ValueWithContext[AnyRef]]]])
```

Comment:
I wonder if we need additional transformer, which handles situation when we don't evict JOINED, but we always take last one - currently there is simplified implementation in EnrichWithAdditionalDataTransformer in DevConfigCreator. I know it can be emulated with "inifite" window length, and "last" aggregation - but it's not v. elegant?

Reply:
IMO even if we will have fixed last synchronization we should always have some time-to-leave (window) to be sure that we don't have some leak of resources... but by this comment you want to say that I should add some abstraction with such will be easier to achieve this without too much code duplication?

Reply:
Don't know TBH... we can leave it as it is now...